### PR TITLE
full screen gameplay, player score persists

### PR DESCRIPTION
**Pull request by arteartemis**

### CODE REVIEW

File: public/js/Bullet.js [hunk 1]
```diff
@@ -65,7 +68,11 @@ class Bullet extends GameObject {
   draw(ctx: Object) {
     ctx.fillStyle = this._color;
     ctx.beginPath();
-    ctx.arc(this._x, this._y, this._size, 0, 2*Math.PI);
+    ctx.arc(
+      this._x * Globals.widthRatio,
+      this._y * Globals.heightRatio,
+      this._size * Globals.widthRatio,
+      0, 2*Math.PI);
```
Comment (reparadocs) [inline public/js/Bullet.js]: nit: ')' on next line

File: public/js/GameObject.js [hunk 2]
```diff
@@ -69,7 +72,10 @@ class GameObject {
 
   draw(ctx: Object) {
     ctx.fillStyle = this._color;
-    ctx.fillRect(this._x - (this._width / 2), this._y - (this._height / 2),this._width, this._height);
+    ctx.fillRect(
+      (this._x * Globals.widthRatio) - ((this._width * Globals.widthRatio) / 2),
+      (this._y * Globals.heightRatio) - ((this._height * Globals.heightRatio) / 2),
+      (Globals.widthRatio * this._width), (Globals.heightRatio * this._height));
```
Comment (reparadocs) [inline public/js/GameObject.js]: Nit: ')' on next line

File: public/js/Bullet.js (hunk 1 above)
Comment (reparadocs) [inline public/js/Bullet.js]: Nit: ')' on next line

File: public/js/Keys.js [hunk 3]
```diff
@@ -47,15 +49,16 @@ class Keys {
...
+		const diffY = this._mouseY - this._localPlayer.getY() * Globals.heightRatio;
 		const length = Math.sqrt(diffX * diffX + diffY * diffY);
 		const vector = [diffX / length, diffY / length];
 		this._dir = vector;
 	}
 
 	isPlayerOnMouse() {
-		return Math.abs(this._localPlayer.getX() - this._mouseX) < 15 && Math.abs(this._localPlayer.getY() - this._mouseY) < 15;
+		return Math.abs(this._localPlayer.getX() * Globals.widthRatio - this._mouseX) < 15
+		&& Math.abs(this._localPlayer.getY() * Globals.heightRatio - this._mouseY) < 15;
```
Comment (reparadocs) [inline public/js/Keys.js]: Nit: indent here

File: public/js/Player.js [hunk 4]
```diff
@@ -104,29 +105,39 @@ class Player extends GameObject {
 	}
 
 	draw(ctx: Object) {
+		var x_pos = this._x * Globals.widthRatio;
```
Comment (reparadocs) [inline public/js/Player.js]: nit: use const instead of var

File: public/js/Player.js [hunk 5]
```diff
@@ -104,29 +105,39 @@ class Player extends GameObject {
...
+		var y_pos = this._y * Globals.heightRatio;
```
Comment (reparadocs) [inline public/js/Player.js]: nit: use const instead of var

File: public/js/Player.js [hunk 6]
```diff
@@ -104,29 +105,39 @@ class Player extends GameObject {
...
 			if (size < 1) {
 				size = 1;
 			}
-			ctx.arc(this._x, this._y, size, 0, 2*Math.PI);
-			ctx.moveTo(this._x + (size * this._shootDir[0]), this._y + (size * this._shootDir[1]));
-			ctx.lineTo(this._x + ((size - 5) * this._shootDir[0]), this._y + ((size - 5) * this._shootDir[1]));
+			ctx.arc(x_pos, y_pos, size * Globals.widthRatio, 0, 2*Math.PI);
+			ctx.moveTo(
+				x_pos + (size * Globals.widthRatio * this._shootDir[0]),
+				y_pos + (size * Globals.widthRatio * this._shootDir[1]));
```
Comment (reparadocs) [inline public/js/Player.js]: nit: ')' on next line

File: public/js/Player.js [hunk 7]
```diff
@@ -104,29 +105,39 @@ class Player extends GameObject {
...
+			ctx.lineTo(
+				x_pos + ((size * Globals.widthRatio - 5) * this._shootDir[0]),
+				y_pos + ((size * Globals.widthRatio - 5) * this._shootDir[1]));
```
Comment (reparadocs) [inline public/js/Player.js]: nit: ')' on next line

File: public/js/Player.js [hunk 8]
```diff
@@ -104,29 +105,39 @@ class Player extends GameObject {
...
 			ctx.stroke();
 		}
 
-		ctx.font = "12px serif";
-		ctx.fillText(this._bulletCount, this._x - 5, this._y + 5);
+		ctx.font = "18px serif";
+		ctx.fillText(
+			this._bulletCount, 
+			this._x * Globals.widthRatio - 5,
+			this._y * Globals.heightRatio + 5);
```
Comment (reparadocs) [inline public/js/Player.js]: nit: ')' on next line

File: public/js/Player.js [hunk 9]
```diff
@@ -152,7 +163,7 @@ class Player extends GameObject {
 		this._y = Math.round(Math.random()*(Constants.gameHeight-40)) + 20;
 		this._bulletCount = 1;
 		this._alive = true;
-		this._score = 0;
+		//this._score = 0;
```
Comment (reparadocs) [inline public/js/Player.js]: Remove it if not necessary (but why are we removing score?)

File: public/js/Resource.js [hunk 10]
```diff
@@ -19,7 +20,11 @@ class Resource extends GameObject {
   draw(ctx: Object) {
     ctx.fillStyle = this._color;
     ctx.beginPath();
-    ctx.arc(this._x, this._y, Constants.resourceSize, 0, 2*Math.PI);
+    ctx.arc(
+      this._x * Globals.widthRatio,
+      this._y * Globals.heightRatio,
+      Constants.resourceSize * Globals.widthRatio,
+      0, 2*Math.PI);
```
Comment (reparadocs) [inline public/js/Resource.js]: Nit ')' on next line

File: public/js/game.js [hunk 11]
```diff
@@ -32,6 +32,9 @@ function init() {
 	canvas.width = window.innerWidth;
 	canvas.height = window.innerHeight;
 
+	Constants.widthRatio = window.innerWidth / Constants.gameWidth;
```
Comment (reparadocs) [inline public/js/game.js]: If they're constants, they shouldn't change, put this in Globals

File: public/js/game.js [hunk 12]
```diff
@@ -272,6 +278,8 @@ function update() {
 function draw() {
 	// Wipe the canvas clean
 	ctx.clearRect(0, 0, canvas.width, canvas.height);
+	ctx.fillStyle = '#FFC0CB';
```
Comment (reparadocs) [inline public/js/game.js]: Put this color in Constants

File: public/js/Bullet.js (hunk 1 above)
Comment (reparadocs) [reply public/js/Bullet.js]: Also add a ',' after this line, and do this for every time you move the ')' down

**Review by reparadocs — CHANGES_REQUESTED** [review]
Overall good, needs some minor changes and fix merge conflict with the yarn file

File: public/js/Player.js (hunk 9 above)
Comment (arteartemis) [reply public/js/Player.js]: we aren't removing it -- we're just not resetting the player's score in between deaths (score persists)

File: public/js/ClientPlayer.js [hunk 13]
```diff
@@ -8,8 +8,8 @@ class ClientPlayer extends Player {
   _offset: number;
   _borders: Array<Object>;
 
-  constructor(startX: number, startY: number, dir: Array<number>, color: string, createBullet: Function) {
-    super(startX, startY, dir, color, createBullet);
+  constructor(startX: number, startY: number, dir: Array<number>, color: string, name: string, createBullet: Function) {
```
Comment (reparadocs) [inline public/js/ClientPlayer.js]: Keep lines to width 80 max:

`
constructor(

    startX: number,
 
    startY: number, 

    dir: Array<number>,

    color: string, 

    name: string, 

    createBullet: Function,

  ) {
`

File: public/js/Constants.js [hunk 14]
```diff
@@ -14,8 +14,18 @@ var Constants: {
 
   resourceSize: number,
   numResources: number,
-
```
Comment (reparadocs) [inline public/js/Constants.js:17]: keep lines in between to differentiate different types of constants (resource constants are different than history constants, etc.)

File: public/js/Constants.js [hunk 15]
```diff
@@ -14,8 +14,18 @@ var Constants: {
...
   maxHistoryBuffer: number,
+
+  animals: Array<string>,
+  adjectives: Array<string>,
```
Comment (reparadocs) [inline public/js/Constants.js]: name_prefix_adjectives might be more descriptive to reduce confusion in the future

File: public/js/Constants.js [hunk 16]
```diff
@@ -14,8 +14,18 @@ var Constants: {
 
   resourceSize: number,
   numResources: number,
-
   maxHistoryBuffer: number,
+
+  animals: Array<string>,
```
Comment (reparadocs) [inline public/js/Constants.js]: name_suffix_animals might be more descriptive to reduce confusion in the future

File: public/js/Constants.js [hunk 17]
```diff
@@ -34,6 +44,28 @@ var Constants: {
   resourceSpawnRate: 60,
 
   maxHistoryBuffer: 500,
+
+  animals: ['alligator', 'ant', 'antelope', 'bat', 'bear', 'beaver', 'bee', 'beetle',
```
Comment (reparadocs) [inline public/js/Constants.js]: would be better if you can figure out a way to pull this from a separate text file but if it takes too much effort, don't worry about it not a big deal

File: public/js/Constants.js [hunk 18]
```diff
@@ -34,6 +44,28 @@ var Constants: {
...
+  'bluebird', 'buffalo', 'bunny', 'butterfly', 'bull', 'camel', 'cat', 'chicken',
+  'chimpanzee', 'chipmunk', 'cow', 'coyote', 'crab', 'crane', 'crocodile', 'deer',
+  'dog', 'dolphin', 'donkey', 'dove', 'dragon', 'eagle', 'elk', 'eel', 'fish', 'flamingo',
+  'fly', 'fox', 'frog', 'giraffe', 'goat', 'goose', 'gorilla', 'horse', 'hummingbird',
+  'jaguar', 'koala', 'ladybug', 'lamb', 'lizard', 'llama', 'lobster', 'monkey', 'moose',
+  'moth', 'mouse', 'ostrich', 'rabbit', 'rat', 'robin', 'rooster', 'seahorse', 'shark',
+  'sheep', 'toad', 'unicorn','zebra'],
+
+  adjectives: ['incredible', 'awesome', 'amazing', 'marvelous', 'wonderful',
```
Comment (reparadocs) [inline public/js/Constants.js]: same as above

File: public/js/Keys.js [hunk 19]
```diff
@@ -58,7 +58,7 @@ class Keys {
 
 	isPlayerOnMouse() {
 		return Math.abs(this._localPlayer.getX() * Globals.widthRatio - this._mouseX) < 15
-		&& Math.abs(this._localPlayer.getY() * Globals.heightRatio - this._mouseY) < 15;
+			&& Math.abs(this._localPlayer.getY() * Globals.heightRatio - this._mouseY) < 15;
```
Comment (reparadocs) [inline public/js/Keys.js]: Hmm, check out your tabs... the settings should be converting tabs to spaces and set it to 2 spaces

File: public/js/Player.js [hunk 20]
```diff
@@ -14,15 +14,37 @@ class Player extends GameObject {
 	_score: number;
 	_shootDir: Array<number>;
 	_createBullet: Function;
+	_name: string;
+	_init_name: boolean;
 
-	constructor(startX: number, startY: number, dir: Array<number>, color: string, createBullet: Function) {
+	constructor(startX: number, startY: number, dir: Array<number>, color: string, name: string, createBullet: Function) {
```
Comment (reparadocs) [inline public/js/Player.js]: try to limit to 80 characters/line

File: public/js/Player.js [hunk 21]
```diff
@@ -14,15 +14,37 @@ class Player extends GameObject {
...
 		this._shootDir = dir;
 		this._bulletCount = 1;
 		this._chargeTime = 0;
 		this._createBullet = createBullet;
 		this._score = 0;
+		this._init_name = false;
+		this._name = this._generateName(name);
+	}
+
+	_generateName(name: string) {
```
Comment (reparadocs) [inline public/js/Player.js]: Check tabs again for this whole file

File: public/js/Player.js [hunk 22]
```diff
@@ -14,15 +14,37 @@ class Player extends GameObject {
...
+		if (name != null) {
+			return name;
+		}
+
+		if (!this._init_name) {
```
Comment (reparadocs) [inline public/js/Player.js]: Hmm, why do you need __init__name? Can't you just check if this._name is null or something along those lines? Seems messy to have another bool for this

File: public/js/game.js [hunk 23]
```diff
@@ -307,9 +317,26 @@ function draw() {
...
+	ctx.fillStyle = Constants.color_black;
+	ctx.font = Constants.font_size_user_stats;
+	ctx.fillText("Name: " + localPlayer.getName(), 10, canvas.height - 50);
+	ctx.fillText("Shots: " + localPlayer.getBulletCount(), 10, canvas.height - 30);
+	ctx.fillText("Score: " + localPlayer.getScore(), 10, canvas.height - 10);
+
+	var clonedRemote = remotePlayers.slice(0);
+	clonedRemote.push(localPlayer);
+	clonedRemote.sort(function(a, b) {
+    return b._score - a._score;
```
Comment (reparadocs) [inline public/js/game.js:341]: fix tabbing here

File: public/js/game.js [hunk 24]
```diff
@@ -307,9 +317,26 @@ function draw() {
...
+	});
+
+	ctx.font = Constants.font_size_scoreboard;
+	for (i = 0; i < (clonedRemote.length > 5 ? 5 : clonedRemote.length); i++) {
+		ctx.fillText(
```
Comment (reparadocs) [inline public/js/game.js]: Fix tabs-> 2 spaces formatting

File: node_modules/.yarn-integrity [hunk 25]
```diff
@@ -1 +1 @@
-2e6bdee5b01093c3fb20daeaae838b7c24e86ee68a22f5a1670a583b61581bdf
```
Comment (reparadocs) [inline node_modules/.yarn-integrity]: Try to add the whole node_modules dir to .gitignore. If it takes you longer than 5 mins, lemme know and I'll do it

**Review by reparadocs — CHANGES_REQUESTED** [review]
really good overall, fix your editor's settings to change tabs to 2 spaces and reformat all the files (there should be a menu option that lets you easily reformat files with your new setting), and a few minor formatting nits

**Pull request merged**